docs/components/receive.md

```diff
@@ -95,6 +95,27 @@ The example content of `hashring.json`:
 
 With such configuration any receive listens for remote write on `<ip>10908/api/v1/receive` and will forward to correct one in hashring if needed for tenancy and replication.
 
+### AZ-aware Ketama hashring (experimental)
+
+In order to ensure even spread for replication over nodes in different availability-zones, you can choose to indicate AZ as part of the endpoint. If we for example have a 6 node cluster, spread over 3 different availability zones: A, B and C, we could use the following example `hashring.json`:
+
+```json
+[
+    {
+        "endpoints": [
+            "127.0.0.1:10907,A",
+            "127.0.0.1:11907,B",
+            "127.0.0.1:12907,C",
+            "127.0.0.1:13907,A",
+            "127.0.0.1:14907,B",
+            "127.0.0.1:16907,C"
+        ]
```

Comment:
Good point. With current code it would treat all non-AZ endpoints as being in an az called `""` so things would technically work, but probably not like the user had intended.

I would personally lean towards throwing an error immediately on startup if there is a mix between AZ and non AZ endpoints as at least I can't see the use-case for doing that intentionally. WDYT?

Reply:
SGTM. Maybe we can use a separate field for az endpoints, as already suggested, in the hashring for that and make it mutual exclusive with `endpoints`? I think it would make checks simpler and we don't have to worry about backwards compatibility.

Reply:
I added extra validation both for the split AZ/non AZ configuration, and also az configuration + hashed algorithm. 

I personally felt that adding the extra field didn't add much value since I still had to perform validation on it afterwards, so I skipped that for now. How do you feel about the current state of the implementation?

Reply:
I think the extra field would make string splitting and checks for mixed endpoints simpler/obsolete and in my opinion it would help making the code more comprehendible.

Reply:
After discussion on slack, we've decided to await feedback from others before proceeding with making further changes here

Reply:
Having a separate object with named fields would be indeed nice, but there's no good way to version the configuration file now.

Reply:
How hard would it be to make `endpoints` polymorphic, so that it can be either a slice of strings or slice of objects? We could then try parsing the slice of strings first, and if it fails we would fall back to parsing a slice of objects. The object would contain the IP and the AZ and it would allow us to add multiple fields with strong typing in the future.

Reply:
+1 to this! I guess you'd probably need to modify the `HashringConfig` and then parse endpoints via a type switch.